tests/e2e/pubsub/pubsub_test.go

```diff
@@ -0,0 +1,189 @@
+// +build e2e
+
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package pubsubapp_e2e
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"os"
+	"reflect"
+	"testing"
+
+	
+	"github.com/dapr/dapr/tests/e2e/utils"
+	kube "github.com/dapr/dapr/tests/platforms/kubernetes"
+	"github.com/dapr/dapr/tests/runner"
+	"github.com/stretchr/testify/require"
+)
+
+const numHealthChecks = 3      // Number of get calls before starting tests.
+const testManyEntriesCount = 5 // Anything between 1 and the number above (inclusive).
+
+var tr *runner.TestRunner
+
+const (
+	publisherAppName  = "pubsub-publisher"
+	subscriberAppName = "pubsub-subscriber"
+)
+
+type testCommandRequest struct {
+	Message string `json:"message,omitempty"`
+}
+
+// sent to the publisher app, which will publish data to dapr
+type publishCommand struct {
+	Topic string `json:"topic"`
+	Data  string `json:"data"`
+}
+
+type appResponse struct {
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+// data returned from the subscriber app
+type receivedMessagesResponse struct {
+	ReceivedByTopicA []string `json:"pubsub-a-topic"`
+	ReceivedByTopicB []string `json:"pubsub-b-topic"`
+}
+
+var receivedMessages []string
+
+// indexHandler is the handler for root path
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Printf("indexHandler is called\n")
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(appResponse{Message: "OK"})
+}
+
+// sends messages to the publisher app.  The publisher app does the actual publish
+func sendToPublisher(t *testing.T, publisherExternalURL string, topic string) ([]string, error) {
+	var sentMessages []string
+	commandBody := publishCommand{Topic: topic}
+	offset := rand.Intn(100)
+	for i := offset; i < offset+100; i++ {
```

Comment:
Is this 100 the same 100 as above? If so, share the same const. If not, create a new const.

Reply:
Good point, actually not the same :)